201-userdefined-routes-appliance/azuredeploy.json

```diff
@@ -0,0 +1,553 @@
+            }
+        },
+		"dnsNameForPubIP1": {
+            "type": "string",
+            "metadata": {
+                  "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+            }
+        },
+		"dnsNameForPubIP2": {
+            "type": "string",
+            "metadata": {
+                  "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+            }
+        },
+        "dnsNameForPubIP3": {
+            "type": "string",
+            "metadata": {
+                  "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+            }
+        },
+		"vNetAddressPrefix":{
+			"type" : "string",
+			"defaultValue" : "10.0.0.0/16",
+			"metadata" : {
+			  "Description" : "Address prefix"
+			}
+		},
+		"subnet1Prefix" : {
+			"type" : "string",
+			"defaultValue" : "10.0.0.0/24",
+			"metadata" : {
+			  "Description" : "Subnet 1 Prefix"
+			}
+		},
+		"subnet2Prefix" : {
+			"type" : "string",
+			"defaultValue" : "10.0.1.0/24",
+			"metadata" : {
+          "Description" : "Subnet 2 Prefix"
+			}
+		},
+		"subnet3Prefix" : {
+			"type" : "string",
+			"defaultValue" : "10.0.2.0/24",
+			"metadata" : {
+			"Description" : "Subnet 3 Prefix"
+			}
+		},
+		"publicIPAddressType": {
+			"type": "string",
+			"defaultValue": "Dynamic",
+			"allowedValues": [
+				"Dynamic",
+				"Static"
+			],
+			"metadata" : {
+			"description" : "Type of public IP address"
+			}
+		},
+		"windowsOSVersion": {
+			"type": "string",
+			"defaultValue": "2012-R2-Datacenter",
+			"allowedValues": [
+			"2008-R2-SP1",
+			"2012-Datacenter",
+			"2012-R2-Datacenter"
+			],
+			"metadata": {
+				"description": "The Windows version for the VM. This will pick a fully patched image of this given Windows version. Allowed values: 2008-R2-SP1, 2012-Datacenter, 2012-R2-Datacenter."
+			}
+		},
+		"PublicIPNameForVM1": {
+			"type": "string",
+			"defaultValue": "FEVM1_PIP",
+			"metadata": {
+				"description" : "Name of the Public IP"
+			}
+		},
+		"PublicIPNameForVM2": {
+			"type": "string",
+			"defaultValue": "NVAVM1_PIP",
+			"metadata": {
+				"description" : "Name of the Public IP"
+			}
+		},
+		"PublicIPNameForVM3": {
+			"type": "string",
+			"defaultValue": "BEVM1_PIP",
+			"metadata": {
+				"description" : "Name of the Public IP"
+			}
+		},
+		"VNetName": {
+			"type": "string",
+			"defaultValue": "",
+			"metadata": {
+				"description" : "Name of the VNet"
+			}	
+		},
+		"Subnet1Name": {
+			"type": "string",
+			"defaultValue": "",
+			"metadata": {
+				"description" : "Name of the Subnet"
+			}	
+		},
+		"Subnet2Name": {
+			"type": "string",
+			"defaultValue": "",
+			"metadata": {
+				"description" : "Name of the Subnet"
+			}	
+		},
+		"Subnet3Name": {
+			"type": "string",
+			"defaultValue": "",
+			"metadata": {
+				"description" : "Name of the Subnet"
+			}	
```

Comment:
Consider making VNET, subnet, PublicIP names above variables instead of parameters to reduce the number of parameters required to run this template